src/libcadet/model/parts/ConvectionDispersionOperator.cpp

```diff
@@ -185,29 +186,25 @@ bool ConvectionDispersionOperatorBase::configure(UnitOpIdx unitOpIdx, IParameter
  */
 bool ConvectionDispersionOperatorBase::notifyDiscontinuousSectionTransition(double t, unsigned int secIdx)
 {
-	double prevVelocity = static_cast<double>(_curVelocity);
-
 	// If we don't have cross section area, velocity is given by parameter
+	const int dirOld = _dir;
 	if (_crossSection <= 0.0)
-		_curVelocity = getSectionDependentScalar(_velocity, secIdx);
+	{
+		const double dirNew = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
+		_dir = (dirNew >= 0.0) ? 1 : -1;
+	}
```

Comment:
We also need to set `_curVelocity` here. In `setFlowRates()`, `_curVelocity` is only set if `_crossSection > 0.0`. Thus, we need to set `_curVelocity` here (inside the `curVelocity <= 0.0` branch) in this case. Otherwise, it will be undefined.

Reply:
I think you're right, but I'm also a bit confused. ^^
> (inside the curVelocity <= 0.0 branch) 

Do you mean inside the _crossSection <= 0.0 branch?

However, a bit up, you also write
> However, the flipping must occur only in the else if branch above. Otherwise, this could happen:
Constant flow rate per section, transition from positive to negative.

With that you mean that it was simply wrong to put the statement that overwrites _dir outside the conditions but it's true for both cases, right?

If so, doesn't this mean, I can combine both branches with a simple OR (`||`)?

Reply:
> Do you mean inside the _crossSection <= 0.0 branch?

Yes, sorry. I meant the `_crossSection <= 0.0` branch.

> With that you simply meant that it was wrong to put the statement that overwrites _dir outside the conditions but it needs to be set in both cases, right?

`_dir` needs to be set in all cases. But the flip of `_curVelocity` must only happen in the `else if` branch:
```cpp
else if (!_velocity.empty())
{
	// ...

	const double dirNew = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
	_dir = (dirNew >= 0.0) ? 1 : -1;

	if (dirOld * _dir < 0.0)
		_curVelocity *= -1.0;
}
```